### PR TITLE
fix(analytics): Init analytics for the web welcome page.

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

**Comment by bgrozev** [comment]
If we have a bug in lib-jitsi-meet#initLib() which prevents it from running on web (how does the lib get initialized on web anyway?), then obviously we should fix it.

Let me know if you want to invoke the procedure for merging blockers.

**Comment by hristoterezov** [comment]
lib-jitsi-meet#initLib() is currently not executed for web. This logic for web lives in conference.init() and it is slightly different.

One day web and mobile will share this code and than we'll be able to just remove the code that I've added.

I don't believe we should invoke "the procedure for merging blockers" for fixing lib-jitsi-meet#initLib() right after merging this PR. And also fixing lib-jitsi-meet#initLib() could potentially require huge refactoring and a lot of effort which should be separate task that should be separately planned. This refactoring is anyway part of our effort of reactifying Jitsi Meet that is already planned and I'm not sure if refactoring around lib-jitsi-meet#initLib()  is the best next step.

**Comment by bgrozev** [comment]
OK, I'll have to defer to someone else, because I don't understand how any of this works.

**Comment by virtuacoplenny** [comment]
Do you know why the circular calls happen? What is the call chain? Is there no way to sidestep the check for `typeof APP !== 'undefined'` in initLib?

**Comment by virtuacoplenny** [comment]
Had some thoughts..could this not be done at a higher level, like at APP.web? It could even be done right after the store gets made potentially? The initialization of the jitsi-meet (and some error handlers) aren't necessarily bound to APP.conference.init are they? By fixing it at the welcome page other pages that don't go through conference initialization are subject to the same issue.

**Comment by hristoterezov** [comment]
The middleware of unsupported browser is involved in this circular calls but I haven't investigated it.

There is a way to sidestep the check for typeof APP !== 'undefined' in initLib and actually this is how you can reproduce the circular calls.

**Comment by virtuacoplenny** [comment]
I meant side step withouy a circular call. So to confirm, are you going to investigate the cause at least or investigate putting the init higher up?

**Comment by hristoterezov** [comment]
I don't plan to investigate it, that's why I put the code in WelcomePage component.

>  investigate putting the init higher up

I didn't get this. What do you mean?

**Comment by virtuacoplenny** [comment]
Putting the JitsiMeetJs init call at a parent of WelcomePage so other pages, like unsupported browser, also can get analytics.

File: react/features/welcome/components/WelcomePage.web.js [hunk 1]
```diff
@@ -66,6 +68,17 @@ class WelcomePage extends AbstractWelcomePage {
      * @returns {void}
      */
     componentDidMount() {
+        // FIXME: This is not the best place for this logic. Ideally we should
+        // use features/base/lib-jitsi-meet#initLib() action for this use case.
+        // But currently lib-jitsi-meet#initLib()'s logic works for mobile only
+        // (on web it ends up with infinite loop over initLib).
```
Comment (lyubomir) [inline react/features/welcome/components/WelcomePage.web.js]: I'm not reviewing this PR, I'm only mentioning concerns after hearing about this PR in today's morning standup:

1. If it's limited to Web only, it's fine with mobile. But when we're developing we should always account for our end-of-year goal to have Web and mobile merged as much as possible. I'll presume that's been accounted for here and the result is guided by higher priorities.

2. If it ever moves to mobile, please note that the WelcomePage is optional, we have a production use case in which the WelcomePage is disabled, and the WelcomePage only knows about the default domain (which is also configurable in Settings). As this PR currently only does Web-specific modifications, that's fine for now.

**Pull request merged**